extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -540,6 +600,21 @@ export class AmpAutocomplete extends AMP.BaseElement {
     return renderPromise;
   }
 
+  /**
+   * Detaches the children nodes of the given element and returns them as an array.
+   * @param {!Element} element
+   * @return {!Array<!Element>}
+   * @private
+   */
+  getChildNodesAsArray_(element) {
+    const children = [];
+    let child;
+    while ((child = element.firstChild)) {
+      children.push(element.removeChild(child));
+    }
+    return children;
```

Comment:
Do we need to explicitly detach them since they'll be reparented shortly anyways? If not, you could use `fromIterator()` or `[].slice.call(element.children)`.

Reply:
Actually looks like you can just a for loop instead of map() above.

Reply:
When I did just the for loop, the `appendChild` line had the effect of iterating over every other child due to the changing array size. `[].slice.call` did the trick here, thanks for the tip!